src/Features/CSharp/Portable/GoToDefinition/CSharpGoToDefinitionSymbolService.cs

```diff
@@ -20,5 +21,90 @@ public CSharpGoToDefinitionSymbolService()
 
         protected override ISymbol FindRelatedExplicitlyDeclaredSymbol(ISymbol symbol, Compilation compilation)
             => symbol;
+
+        protected override int? GetTargetPositionIfControlFlow(SyntaxToken token)
+        {
+            var node = token.Parent;
+            if (node is null)
+            {
+                return null;
+            }
+
+            switch (token.Kind())
+            {
+                case SyntaxKind.ContinueKeyword:
+                    var foundContinuedLoop = TryFindContainingLoop(node);
+
+                    return foundContinuedLoop?.IsContinuableConstruct() == true
+                        ? foundContinuedLoop.GetFirstToken().Span.Start
+                        : null;
+
+                case SyntaxKind.BreakKeyword:
+                    var foundBrokenLoop = TryFindContainingLoop(node);
+
+                    return foundBrokenLoop?.IsBreakableConstruct() == true
+                        ? foundBrokenLoop.GetLastToken().Span.End
+                        : null;
+
+                case SyntaxKind.YieldKeyword:
+                    var nextToken = token.GetNextToken();
+                    switch (nextToken.Kind())
+                    {
+                        case SyntaxKind.ReturnKeyword:
+                            return GetTargetPositionIfControlFlow(nextToken);
+
+                        case SyntaxKind.BreakKeyword:
+                            var foundReturnableConstruct = TryFindContainingReturnableConstruct(node);
+                            return foundReturnableConstruct?.GetLastToken().Span.End;
+
+                        default:
+                            return null;
+                    }
+
+                case SyntaxKind.ReturnKeyword:
+                    {
+                        var foundReturnableConstruct = TryFindContainingReturnableConstruct(node);
+                        return foundReturnableConstruct?.GetLastToken().Span.End;
+                    }
+
+                default:
+                    return null;
+            }
+
+            SyntaxNode? TryFindContainingLoop(SyntaxNode? node)
+            {
+                while (node?.IsContinuableConstruct() == false && node?.IsBreakableConstruct() == false)
+                {
+                    var kind = node.Kind();
+
+                    if (node.IsReturnableConstruct() ||
+                        SyntaxFacts.GetTypeDeclarationKind(kind) != SyntaxKind.None)
+                    {
+                        return null;
+                    }
+
+                    node = node.Parent;
```

Comment:
i think this will infinite loop on a top level `continue;` statement.  please test.

please add null check for `node` in the while loop.